scripts/react_native_pods.rb

```diff
@@ -3,7 +3,12 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-def use_react_native! (options={})
+def resolve_module(request)
+  script = "console.log(path.dirname(require.resolve('#{request}/package.json')));"
+  Pod::Executable.execute_command('node', ['-e', script], true).strip
```

Comment:
Do we need to run this here? I wonder if there's a way for Ruby to give us `pwd` or something and work our way out of there?

@alloy

Reply:
Alternatively, why don't we detect when `pod install` was run with `--project-directory` flag?

Reply:
We can also do something like here: https://github.com/facebook/react-native/blob/355239b99125ffc7d17009a1c71e2fc42f7c104a/template/ios/Podfile#L7-L9

`config` from `use_native_modules!` is https://github.com/react-native-community/cli/blob/master/packages/platform-ios/native_modules.rb#L124

Maybe it could return something like a path to `iOS` folder?

Reply:
Anyway, this might be better solution because `react-native-codegen` package might be hoisted outside of project root or something.